multiuser/api/che-multiuser-api-resource/src/main/java/org/eclipse/che/multiuser/resource/api/workspace/LimitsCheckingWorkspaceManager.java

```diff
@@ -165,13 +169,32 @@ public WorkspaceImpl updateWorkspace(String id, Workspace update)
   }
 
   @VisibleForTesting
-  void checkMaxEnvironmentRam(WorkspaceConfig config) throws ServerException {
+  void checkMaxEnvironmentRam(Workspace workspace) throws ServerException {
+    if (workspace.getConfig() != null) {
+      checkMaxEnvironmentRam(workspace.getId(), workspace.getConfig());
+    } else {
+      LOG.warn(
+          "Can not estimate memory needed for devfile based workspace {}. Set environment memory limits may work incorrectly",
+          workspace.getId());
+    }
+  }
+
+  @VisibleForTesting
+  void checkMaxEnvironmentRam(@Nullable String workspaceId, WorkspaceConfig config)
+      throws ServerException {
     if (maxRamPerEnvMB < 0) {
       return;
     }
     if (config.getEnvironments().isEmpty()) {
       return;
     }
+
+    if (SidecarToolingWorkspaceUtil.isSidecarBasedWorkspace(config.getAttributes())) {
+      LOG.warn(
```

Comment:
How about: ``"Cannot estimate the memory requirements of a devfile-based workspace{}. The memory limits may not be applied correctly."``